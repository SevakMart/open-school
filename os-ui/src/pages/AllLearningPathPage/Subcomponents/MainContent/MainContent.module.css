.mainContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 100%;
}
.mainContainer .learningPathContainer {
  position: absolute;
  top: 100px;
  left: 400px;
  width: 65%;
  overflow: hidden;
}/*# sourceMappingURL=MainContent.module.css.map */