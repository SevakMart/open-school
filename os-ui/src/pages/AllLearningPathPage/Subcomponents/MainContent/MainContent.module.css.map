{"version":3,"sources":["MainContent.module.scss","../../../../styles/Mixins/FlexMixins.scss","MainContent.module.css"],"names":[],"mappings":"AAEA;ECDE,aAAA;EACA,mBDCqB;ECArB,mBAAA;EACA,8BDDkC;EAChC,WAAA;AEEJ;AFAI;EACI,kBAAA;EACA,UAAA;EACA,WAAA;EACA,UAAA;EACA,gBAAA;AEER","file":"MainContent.module.css"}