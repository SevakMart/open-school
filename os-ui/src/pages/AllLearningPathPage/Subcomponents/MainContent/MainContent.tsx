import FilterComponent from '../FilterComponent/FilterComponent';
import CourseContentHeader from '../CourseCountentHeader/CourseContentHeader';
import styles from './MainContent.module.scss';

const MainContent = () => {
  // const {} = styles;
  const a = 1;
  return (
    <div>
      <FilterComponent />
      <div>
        <CourseContentHeader />
      </div>
    </div>
  );
};
export default MainContent;
