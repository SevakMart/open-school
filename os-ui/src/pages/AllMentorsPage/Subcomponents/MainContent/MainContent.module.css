/*# sourceMappingURL=MainContent.module.css.map */