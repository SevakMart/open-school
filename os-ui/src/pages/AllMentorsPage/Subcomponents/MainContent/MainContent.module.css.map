{"version":3,"sources":[],"names":[],"mappings":"","file":"MainContent.module.css"}