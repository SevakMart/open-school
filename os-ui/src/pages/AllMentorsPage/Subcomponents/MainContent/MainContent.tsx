import { useState } from 'react';
import AllMentorsPageHeader, { Mentors } from '../AllMentorsPageHeader/AllMentorsPageHeader';
import Content from '../Content/Content';
import SavedMentors from '../SavedMentors/SavedMentors';
import styles from './MainContent.module.scss';

/* eslint-disable max-len */

const MainContent = () => {
  const [focusedHeader, setFocusedHeader] = useState(Mentors.ALL_MENTORS);
  const { mainContentContainer } = styles;
  return (
    <div className={mainContentContainer}>
      <AllMentorsPageHeader changeHeaderFocus={(headerTitle) => setFocusedHeader(headerTitle)} />
      {focusedHeader === Mentors.ALL_MENTORS ? <Content /> : <SavedMentors />}
    </div>
  );
};
export default MainContent;
