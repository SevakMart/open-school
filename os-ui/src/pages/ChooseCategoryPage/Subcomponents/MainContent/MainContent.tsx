import CategoryWithSubcategoriesProfile from '../CategoryWithSubcategoriesProfile/CategoryWithSubcategoriesProfile';
import Loader from '../../../../component/Loader/Loader';
import { SearchedCategoryType } from '../../../../types/SearchedCategoryType';
import { ErrorField } from '../../../../component/ErrorField/ErrorField';
import styles from './MainContent.module.scss';

/* eslint-disable max-len */

const MainContent = ({ isLoading, errorMessage, searchedCategories }:{isLoading:boolean, errorMessage:string, searchedCategories:SearchedCategoryType}) => {
  const { categoriesList } = styles;
  return (
    <>
      <div className={categoriesList}>
        {isLoading && <Loader />}
        {errorMessage !== '' && <ErrorField.MainErrorField className={['subcategoriesErrorField']}>{errorMessage}</ErrorField.MainErrorField>}
        {errorMessage === '' && (
          Object.entries(searchedCategories).map((category) => (
            <CategoryWithSubcategoriesProfile
              key={category[0]}
              parentCategory={category[0]}
              subcategories={category[1]}
            />
          ))
        )}
      </div>
    </>
  );
};
export default MainContent;
