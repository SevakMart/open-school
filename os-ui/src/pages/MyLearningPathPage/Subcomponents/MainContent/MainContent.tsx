import { useSelector } from 'react-redux';
import { RootState } from '../../../../redux/Store';
import { UserCourseType } from '../../../../types/CourseTypes';
import InProgressCourse from '../InProgressCourse/InProgressCourse';
import CompletedCourse from '../CompletedCourse/CompledtedCourse';
import ContentRenderer from '../../../../component/ContentRenderer/ContentRenderer';
import styles from './MainContent.module.scss';

/* eslint-disable max-len */

const MyLearningPathMainContent = () => {
  const userEnrolledCourseByCourseStatus = useSelector<RootState>((state) => state.userEnrolledCourseByCourseStatus);
  const { entity, isLoading, errorMessage } = userEnrolledCourseByCourseStatus as {entity:any[], isLoading:boolean, errorMessage:string};
  const { courseContainer } = styles;
  return (
    <div className={courseContainer}>
      <ContentRenderer
        isLoading={isLoading}
        errorMessage={errorMessage}
        entity={entity}
        errorFieldClassName="myLearningPathsErrorStyle"
        isMyLearningPathPage
        render={(entity) => (
          entity.map((userCourse:UserCourseType, index:number) => {
            if (userCourse.courseStatus === 'IN_PROGRESS') {
              return (
                <InProgressCourse
                  key={index}
                  title={userCourse.title}
                  courseStatus={userCourse.courseStatus}
                  percentage={userCourse.percentage}
                  remainingTime={userCourse.remainingTime}
                  dueDate={userCourse.dueDate}
                  courseId={userCourse.courseId}
                  id={userCourse.courseId}
                />
              );
            }
            return (
              <CompletedCourse
                key={index}
                title={userCourse.title}
                courseStatus={userCourse.courseStatus}
                grade={userCourse.grade}
              />
            );
          })
        )}
      />
    </div>
  );
};
export default MyLearningPathMainContent;
