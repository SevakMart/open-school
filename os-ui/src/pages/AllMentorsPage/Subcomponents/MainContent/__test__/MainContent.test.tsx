import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import MainContent from '../MainContent';

jest.mock('react-i18next', () => ({
  ...jest.requireActual('react-i18next'),
  useTranslation: () => ({
    t: (message:string) => {
      if (message === 'Empty Data Error Message') return 'No data to display';
      return 'Something went wrong please refresh the page';
    },
  }),
}));

describe('Create test cases for MainContent component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MainContent />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
