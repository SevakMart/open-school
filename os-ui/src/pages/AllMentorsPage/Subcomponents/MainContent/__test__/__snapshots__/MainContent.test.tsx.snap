// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for MainContent component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContentContainer"
  >
    <div
      class="mentorHeaderMainContainer"
    >
      <div
        class="navContent"
      >
        <nav>
          <p
            class="activeNav"
          >
            Something went wrong please refresh the page
          </p>
          <p
            class="nonActiveNav"
          >
            Something went wrong please refresh the page
          </p>
        </nav>
      </div>
      <div
        class="search search__allMentorsSearch"
      >
        <div
          style="color: rgb(132, 138, 157); margin-left: 16px;"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
            />
          </svg>
        </div>
        <input
          data-testid="searchInput"
          placeholder="Something went wrong please refresh the page"
          type="text"
          value=""
        />
        <div
          class="closeIcon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7,7 L17,17 M7,17 L17,7"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="mainContent"
    >
      <div
        class="mainContent"
      >
        <div
          class="loader"
        >
          <span />
          <span />
          <span />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
